activation/activation_multi_test.go

```diff
@@ -400,8 +405,13 @@ func Test_Builder_Multi_HappyPath(t *testing.T) {
 		}
 		nipostState[sig.NodeID()] = state
 		tab.mnipost.EXPECT().
-			BuildNIPost(gomock.Any(), sig, ref.PublishEpoch, ref.Hash()).
-			Return(state, nil)
+			BuildNIPost(gomock.Any(), sig, ref.Hash(), gomock.Any()).
+			DoAndReturn(func(_ context.Context, _ *signing.EdSigner, _ types.Hash32,
+				postChallenge *types.NIPostChallenge,
+			) (*nipost.NIPostState, error) {
+				require.Equal(t, postChallenge.PublishEpoch, ref.PublishEpoch, "publish epoch mismatch")
+				return state, nil
+			})
```

Comment:
NIT: parameters passed to the mocked function can be matched directly without having to do define a `DoAndReturn` like this:

```go
		tab.mnipost.EXPECT().BuildNIPost(gomock.Any(), sig, ref.Hash(), gomock.Cond(func(x any) bool {
			postChallenge, ok := x.(*types.NIPostChallenge)
			if !ok {
				return false
			}
			return postChallenge.PublishEpoch == ref.PublishEpoch
		})).Return(state, nil)
```

No need to change the existing test, just an FYI 🙂